src/V3Const.cpp

```diff
@@ -2325,6 +2325,16 @@ class ConstVisitor final : public VNVisitor {
         iterate(nodep);  // Again?
     }
 
+    bool mayConvertToBitwise(AstNodeBiop* const nodep) {
+        if (!nodep->lhsp()->width1()) return false;
+        if (!nodep->rhsp()->width1()) return false;
+        if (!nodep->isPure()) return false;
+        if (nodep->exists([](const AstNodeExpr* const exprp) {
+                return VN_IS(exprp, MethodCall) || VN_IS(exprp, MemberSel);
+            }))
+            return false;
```

Comment:
This goes O(n^2) for large trees.  One option is to compare only to a certain depth.  This conversion is pretty important to performance though, so maybe the memoize trick we did with pure?

Reply:
Yes, you're right. I think we should memoize it. I will do it.

Reply:
I did it.